integration-tests/src/mock.rs

```diff
@@ -0,0 +1,341 @@
+    testing::Header,
+    Storage,
+};
+
+use frame_support::{
+    assert_ok,
+    parameter_types,
+    storage::StorageMap,
+    traits::Everything,
+};
+use frame_support::traits::ConstU32;
+use frame_system as system;
+
+use pallet_permissions::{
+    SpacePermission as SP,
+    SpacePermissions,
+};
+use pallet_posts::{Post, PostUpdate, PostExtension, Comment, Error as PostsError};
+use pallet_reactions::{ReactionId, ReactionKind, Error as ReactionsError};
+use pallet_spaces::{SpaceById, Error as SpacesError};
+use pallet_spaces::types::{SpaceUpdate, SpacesSettings};
+use pallet_space_follows::Error as SpaceFollowsError;
+use pallet_space_ownership::Error as SpaceOwnershipError;
+use pallet_parachain_utils::{
+    mock_functions::*,
+    Error as UtilsError,
+    SpaceId, PostId, User, Content,
+};
+use crate::utils::*;
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<TestRuntime>;
+type Block = frame_system::mocking::MockBlock<TestRuntime>;
+
+frame_support::construct_runtime!(
+        pub enum TestRuntime where
+            Block = Block,
+            NodeBlock = Block,
+            UncheckedExtrinsic = UncheckedExtrinsic,
+        {
+            System: system,
+            Balances: pallet_balances,
+            Timestamp: pallet_timestamp,
+            Permissions: pallet_permissions,
+            Posts: pallet_posts,
+            Profiles: pallet_profiles::{Pallet, Call, Storage},
+            Reactions: pallet_reactions,
+            Roles: pallet_roles,
+            SpaceFollows: pallet_space_follows,
+            SpaceOwnership: pallet_space_ownership,
+            Spaces: pallet_spaces,
+        }
+    );
+
+parameter_types! {
+        pub const BlockHashCount: u64 = 250;
+    }
+
+impl system::Config for TestRuntime {
+    type BaseCallFilter = Everything;
+    type BlockWeights = ();
+    type BlockLength = ();
+    type Origin = Origin;
+    type Call = Call;
+    type Index = u64;
+    type BlockNumber = BlockNumber;
+    type Hash = H256;
+    type Hashing = BlakeTwo256;
+    type AccountId = AccountId;
+    type Lookup = IdentityLookup<Self::AccountId>;
+    type Header = Header;
+    type Event = Event;
+    type BlockHashCount = BlockHashCount;
+    type DbWeight = ();
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = pallet_balances::AccountData<u64>;
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+    type SS58Prefix = ();
+    type OnSetCode = ();
+    type MaxConsumers = ConstU32<16>;
+}
+
+parameter_types! {
+        pub const MinimumPeriod: u64 = 5;
+    }
+
+impl pallet_timestamp::Config for TestRuntime {
+    type Moment = u64;
+    type OnTimestampSet = ();
+    type MinimumPeriod = MinimumPeriod;
+    type WeightInfo = ();
+}
+
+parameter_types! {
+        pub const ExistentialDeposit: u64 = 1;
+    }
+
+impl pallet_balances::Config for TestRuntime {
+    type Balance = u64;
```

Comment:
Isn't there an external `pub(super) type Balance` to place here?

Reply:
no, I copied it as if found in the solochain